google/auth/external_account.py

```diff
@@ -140,6 +143,11 @@ def __init__(
             self._client_auth = None
         self._sts_client = sts.Client(self._token_url, self._client_auth)
 
+        # Set metrics options or create default if options were not provided.
+        self._metrics_options = (
+            metrics_options or self._create_default_metrics_options()
```

Comment:
why not just use the default one?

Reply:
We need to be able to pass the metrics options through the constructor because of the way impersonated credentials are handled in this library. If impersonation is used, the original credential (defined by the ADC file) is created. That credential knows that it is using impersonation because it has an impersonation_url passed in the constructor. Then, the original credential creates an impersonated credential object and passes a source credential to the impersonated credential. The source credential is just a copy of the original credential but with all the impersonation information removed. The source credential is the object that actually makes the call to sts, so we need to have some for it to know that the call to sts is for service account impersonation by passing the metrics object from the original credential. If we just use the create default, source metric has no way to know its being used for impersonation.

Reply:
I see. Thank you for the detailed explanation!